src/core/backbone_router/bbr_manager.cpp

```diff
@@ -505,21 +505,34 @@ NdProxyTable &Manager::GetNdProxyTable(void)
     return mNdProxyTable;
 }
 
-bool Manager::ShouldForwardDuaToBackbone(const Ip6::Address &aAddress)
+bool Manager::ShouldForwardDuaToBackbone(const Ip6::Address &srcAddress, const Ip6::Address &dstAddress)
 {
     bool              forwardToBackbone = false;
     Mac::ShortAddress rloc16;
     Error             error;
+    Child *           child;
+    bool              isMedChild = false;
+    Mac::Address      rloc;
 
     VerifyOrExit(Get<Local>().IsPrimary());
-    VerifyOrExit(Get<Leader>().IsDomainUnicast(aAddress));
+    VerifyOrExit(Get<Leader>().IsDomainUnicast(srcAddress));
+    VerifyOrExit(Get<Leader>().IsDomainUnicast(dstAddress));
 
-    VerifyOrExit(!mNdProxyTable.IsRegistered(aAddress.GetIid()));
+    VerifyOrExit(!mNdProxyTable.IsRegistered(dstAddress.GetIid()));
 
-    error = Get<AddressResolver>().Resolve(aAddress, rloc16, /* aAllowAddressQuery */ false);
+    // Perform address queries on behalf of minimal end device children
+    child      = Get<ChildTable>().FindChild(srcAddress);
+    isMedChild = (child != nullptr && child->GetDeviceMode().IsMinimalEndDevice());
+    error      = Get<AddressResolver>().Resolve(dstAddress, rloc16, isMedChild);
```

Comment:
We didn't send address query here because address query should have already been sent in the mesh forwarder. 

Even if we send Address Query for MED here, it's not helpful for forwarding this message because BBR has to device which whether to forward the message to Backbone before the Address Notification is received.